pkg/artifactory/resource/repository/virtual/virtual.go

```diff
@@ -62,7 +63,7 @@ var BaseVirtualRepoSchema = map[string]*schema.Schema{
 	"project_key": {
 		Type:             schema.TypeString,
 		Optional:         true,
-		ValidateDiagFunc: utils.ProjectKeyValidator,
+		ValidateDiagFunc: validator.ProjectKey,
```

Comment:
I think I like `ProjectKeyValidator` better. Your current name doesn't indicate the type or purpose at all correctly